go/consensus/cometbft/apps/registry/admission.go

```diff
@@ -0,0 +1,237 @@
+package registry
+
+import (
+	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	"github.com/oasisprotocol/oasis-core/go/common/node"
+	"github.com/oasisprotocol/oasis-core/go/consensus/cometbft/api"
+	registryState "github.com/oasisprotocol/oasis-core/go/consensus/cometbft/apps/registry/state"
+	registry "github.com/oasisprotocol/oasis-core/go/registry/api"
+)
+
+// policyFn is an admission policy verification function.
+type policyFn func(
```

Comment:
Do we need new types here? Maybe it would be better to move structs here and add 2 new methods, `Validate` and `Verify`.

Reply:
You probably mean move these functions into `registry/api`. Yeah that actually makes more sense.